lib/c-api/src/wasm_c_api/wat.rs

```diff
@@ -6,11 +6,20 @@ use super::types::wasm_byte_vec_t;
 /// In case of failure, `wat2wasm` returns `NULL`.
 #[cfg(feature = "wat")]
 #[no_mangle]
-pub unsafe extern "C" fn wat2wasm(wat: &wasm_byte_vec_t) -> Option<Box<wasm_byte_vec_t>> {
-    let wat: &[u8] = wat.into_slice()?;
-    let result: wasm_byte_vec_t = c_try!(wasmer::wat2wasm(wat)).into_owned().into();
+pub unsafe extern "C" fn wat2wasm(wat: &wasm_byte_vec_t, out: &mut wasm_byte_vec_t) {
+    let wat: &[u8] = match wat.into_slice() {
+        Some(v) => v,
+        _ => return,
+    };
+    let result: wasm_byte_vec_t = match wasmer::wat2wasm(wat) {
+        Ok(val) => val.into_owned().into(),
+        Err(err) => {
+            crate::error::update_last_error(err);
+            return;
+        }
```

Comment:
How does the user know whether `wat2wasm` succeeded? Is there a way to initialize a `wasm_byte_vec_t` to "clear" which the user can then test for? Or does it always check for errors with a separate "did the last thing fail" call after every call to `wat2wasm`?

Reply:
That's a fair point, in this case I'd assume the user would initialize to null prior and check for that, but perhaps it makes more sense for us to zero the fields for them on failure

Reply:
You can also use the return value if you like.